src/components/auth/signUp/signUpForm1.component.tsx

```diff
@@ -0,0 +1,217 @@
+import React from 'react';
+import {
+  View,
+  ViewProps,
+} from 'react-native';
+import {
+  ThemedComponentProps,
+  ThemeType,
+  withStyles,
+} from '@kitten/theme';
+import {
+  Button,
+  CheckBox,
+} from '@kitten/ui';
+import { ValidationInput } from '@src/components/common';
+
+export interface SignUpForm1Type {
+  firstName: string;
+  lastName: string;
+  date: string;
+  email: string;
+  password: string;
+}
+
+interface ComponentProps {
+  onSubmit: (value: SignUpForm1Type) => void;
+}
+
+export type SignUpForm1Props = ThemedComponentProps & ViewProps & ComponentProps;
+
+interface State {
+  firstName: string;
+  lastName: string;
+  date: string;
+  email: string;
+  password: string;
+  firstNameValid: boolean;
+  lastNameValid: boolean;
+  dateValid: boolean;
+  emailValid: boolean;
+  passwordValid: boolean;
+  termsAccepted: boolean;
+}
+
+class SignUpForm1Component extends React.Component<SignUpForm1Props, State> {
+
+  public state: State = {
+    firstName: '',
+    lastName: '',
+    date: '',
+    email: '',
+    password: '',
+    firstNameValid: false,
+    lastNameValid: false,
+    dateValid: false,
+    emailValid: false,
+    passwordValid: false,
+    termsAccepted: false,
+  };
+
+  private nameValidator: RegExp = /[a-z ,.'-]+/;
```

Comment:
Does it make sense to move such validators to the separate module? For example "core/validators"

Reply:
Yes I've thought about it :)